### PR TITLE
Issue #15214: Migrated section 4.8.2.2 Declared when needed to whole config testing

### PR DESCRIPTION
**Pull request by Zopsss**
#15214 

https://checkstyle.org/google_style.html#a4.8.2.2

>4.8.2.2 Declared when needed 	VariableDeclarationUsageDistance

### CODE REVIEW

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4822declaredwhenneeded/InputDeclaredWhenNeeded.java [hunk 1]
```diff
@@ -85,20 +93,26 @@ public void testMethod4(int arg) {
...
   public void testMethod5() {
     int arg = 7;
     boolean b = true;
     boolean bb = false;
-    if (b) if (!bb) b = false;
+    if (b) {
+      if (!bb) {
+        b = false;
+      }
+    }
```
Comment (Zopsss) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4822declaredwhenneeded/InputDeclaredWhenNeeded.java:111]: config was complaining about missing usage of `{}`

**Review by Zopsss — COMMENTED** [review]
some changes:

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4822declaredwhenneeded/InputDeclaredWhenNeeded.java [hunk 2]
```diff
@@ -135,22 +150,24 @@ public boolean testMethod7(int a) {
...
-    }
+      {
+        c++;
+        b++;
+      }
+      {
+        n++; // DECLARATION OF VARIABLE 'n' SHOULD BE HERE (distance = 2)
+        m++; // DECLARATION OF VARIABLE 'm' SHOULD BE HERE (distance = 3)
+        b++;
+      }
```
[the remark this answers is not in the document]
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4822declaredwhenneeded/InputDeclaredWhenNeeded.java:167]: indentation is only diff and it is ok to fix indentation in this PR, as it is unrelated to target Check.

**Review by romani — APPROVED** [review]
ok to merge

in split view, without spaces changes looks good.

**Review by rdiachenko — APPROVED** [review]
lgtm

**Pull request merged**